### ISSUE TITLE
Bunch of failing Compet-N tests

### ISSUE THREAD

**Issue by fragglet** [chocolate-doom/chocolate-doom#549]
I ran the statcheck tests because I wanted to check for regressions before doing a new release, and discovered a bunch of demos now desync. Not sure when this started.

    doom/fast/f3m4-526.zip                            f3m4-526.lmp   FAIL
    doom2/coop/max/cm11-335.zip                       UV11-335.LMP   FAIL
    doom2/coop/max/cm13-342.zip                       UV13NET2.LMP   FAIL
    doom2/coop/movies/16nmnet2.zip                    16NMNET2.LMP   FAIL
    doom2/coop/movies/23uvnet2.zip                    COOP1-23.LMP   FAIL
    doom2/coop/movies/25u10637.zip                    COOP1-30.LMP   FAIL
    doom2/coop/movies/30cn3947.zip                    30cn-df.lmp    FAIL
    doom2/coop/movies/30facoop.zip                    30cfvi.lmp     FAIL
    doom2/coop/movies/30facoop.zip                    30cf-xv.lmp    FAIL
    doom2/coop/movies/30nm3737.zip                    NM30COP1.LMP   FAIL
    doom2/coop/movies/30nm3737.zip                    NM30COP2.LMP   FAIL
    doom2/coop/movies/30uv2843.zip                    30UV2843.LMP   FAIL
    doom2/coop/nm100s/cx19-112.zip                    cx19-112.lmp   FAIL
    doom2/coop/speed/cs19-054.zip                     CS19-054.LMP   FAIL
    doom2/coop/speed/cs19-416.zip                     UV19NET2.LMP   FAIL
    doom2/fast/f11-1153.zip                           F11-1153.LMP   FAIL
    doom2/fast/f26-1824.zip                           F26-1824.LMP   FAIL
    doom2/fast/fa26-355.zip                           FA26-355.LMP   FAIL
    doom2/fast/fa26-530.zip                           Fa26-530.lmp   FAIL
    doom2/max/lv23-215.zip                            lv23-215.lmp   FAIL
    doom2/max/lv23-404.zip                            LV23-404.LMP   FAIL
    doom2/max/lv26-344.zip                            LV26-344.LMP   FAIL
    doom2/max/lv26-535.zip                            LV26-535.LMP   FAIL
    doom2/max/lv28-322.zip                            LV28-322.LMP   FAIL
    doom2/max/lv28-438.zip                            LV28-438.LMP   FAIL
    doom2/max/lv30-824.zip                            LV30-824.LMP   FAIL
    doom2/misc/d2-tgm.zip                             Ty28-tgm.lmp   FAIL
    doom2/misc/d2-tgm.zip                             Ty24-tgm.lmp   FAIL
    doom2/misc/d2-tgm.zip                             Ty23-tgm.lmp   FAIL
    doom2/misc/d2-tgm.zip                             Ty09-tgm.lmp   FAIL
    doom2/misc/lv09-tgm.zip                           LV09-TGM.LMP   FAIL
    doom2/misc/lv23-tgm.zip                           LV23-TGM.LMP   FAIL
    doom2/misc/nm100s.zip                             nx23-210.lmp   FAIL
    doom2/misc/nm100s.zip                             nx26-255.lmp   FAIL
    doom2/misc/nm100s.zip                             nx28-146.lmp   FAIL
    doom2/misc/nw26-414.zip                           NW26-414.LMP   FAIL
    doom2/movie/1335nm21.zip                          1335nm21.lmp   FAIL
    doom2/movie/1526nm21.zip                          1526NM21.LMP   FAIL
    doom2/movie/16nmnet2.zip                          16NMNET2.LMP   FAIL
    doom2/movie/1827ns21.zip                          1827ns21.lmp   FAIL
    doom2/movie/2018nm21.zip                          2018NM21.LMP   FAIL
    doom2/movie/2026ns21.zip                          2026ns21.lmp   FAIL
    doom2/movie/28nm5103.zip                          28NM5103.LMP   FAIL
    doom2/movie/29nm4850.zip                          29NM4850.LMP   FAIL
    doom2/movie/30nm2956.zip                          30nm2956.lmp   FAIL
    doom2/movie/30nm3012.zip                          30nm3012.lmp   FAIL
    doom2/movie/30nm3314.zip                          30nm3314.lmp   FAIL
    doom2/movie/30nm3413.zip                          30nm3413.lmp   FAIL
    doom2/movie/30nm3740.zip                          30nm3740.lmp   FAIL
    doom2/movie/30nm3841.zip                          30nm3841.lmp   FAIL
    doom2/movie/30nm3951.zip                          30nm3951.lmp   FAIL
    doom2/movie/30nm4209.zip                          30nm4209.lmp   FAIL
    doom2/movie/30nm4618.zip                          30NM4618.LMP   FAIL
    doom2/movie/30nm4931.zip                          30NM4931.LMP   FAIL
    doom2/movie/30nm4949.zip                          30NM4949.LMP   FAIL
    doom2/movie/30nm4958.zip                          30nm4958.lmp   FAIL
    doom2/movie/30nm5253.zip                          30nm5253.lmp   FAIL
    doom2/movie/30nm5405.zip                          30NM5405.LMP   FAIL
    doom2/movie/30nm5839.zip                          30nm5839.lmp   FAIL
    doom2/movie/30ns5600.zip                          30ns5600.lmp   FAIL
    doom2/movie/30ns6155.zip                          30NS6155.LMP   FAIL
    doom2/movie/nm30coop.zip                          NM30COP1.LMP   FAIL
    doom2/movie/nm30coop.zip                          NM30COP2.LMP   FAIL
    doom2/net-x/23uvnet2.zip                          COOP1-23.LMP   FAIL
    doom2/net-x/30uvnet2.zip                          COOP1-30.LMP   FAIL
    doom2/net-x/uv11-335.zip                          UV11-335.LMP   FAIL
    doom2/net-x/uv13net2.zip                          UV13NET2.LMP   FAIL
    doom2/net-x/uv19net2.zip                          UV19NET2.LMP   FAIL
    doom2/nm100s/ns12-209.zip                         NS12-209.LMP   FAIL
    doom2/nm100s/ns23-112.zip                         NS23-112.LMP   FAIL
    doom2/nm100s/ns26-117.zip                         NS26-117.LMP   FAIL
    doom2/nm100s/ns26-125.zip                         NS26-125.LMP   FAIL
    doom2/nm100s/ns26-255.zip                         ns26-255.lmp   FAIL
    doom2/nm100s/ns28-119.zip                         Ns28-119.lmp   FAIL
    doom2/nm100s/ns28-146.zip                         ns28-146.lmp   FAIL
    doom2/nmare/nm12-314.zip                          NM12-314.LMP   FAIL
    doom2/nmare/nm23-112.zip                          NM23-112.LMP   FAIL
    doom2/nmare/nm26-102.zip                          NM26-102.LMP   FAIL
    doom2/nmare/nm26-108.zip                          NM26-108.LMP   FAIL
    doom2/nmare/nm26-158.zip                          nm26-158.lmp   FAIL
    doom2/nmare/nm26-235.zip                          NM26-235.LMP   FAIL
    doom2/pacifist/pa29-221.zip                       pa29-221.lmp   FAIL
    doom2/respawn/re23-157.zip                        RE23-157.LMP   FAIL
    doom2/respawn/re28-225.zip                        RE28-225.LMP   FAIL
    doom2/respawn/re29-353.zip                        Re29-353.lmp   FAIL
    doom2/tyson/ty23-804.zip                          TY23-804.LMP   FAIL
    doom2/tyson/ty292426.zip                          ty292426.lmp   FAIL
    plutonia/max/p32-1202.zip                         P32-1202.LMP   FAIL
    plutonia/max/pl10-354.zip                         PL10-354.LMP   FAIL
    plutonia/max/pl32-941.zip                         PL32-941.LMP   FAIL
    plutonia/nmare/p27n-129.zip                       PL27-129.LMP   FAIL
    plutonia/nmare/pl21-nm.zip                        PL21-NM.LMP    FAIL
    plutonia/respawn/pr32-349.zip                     pr32-349.lmp   FAIL
    plutonia/speed/pl23-131.zip                       pl23-131.lmp   FAIL
    plutonia/speed/pl32-054.zip                       pl32-054.lmp   FAIL
    plutonia/speed/pl32-106.zip                       PL32-106.LMP   FAIL
    pwads/av/max/av32-441.zip                         av32-441.lmp   FAIL
    pwads/av/nmare/an24-300.zip                       an24-300.lmp   FAIL
    pwads/av/nmare/an26-358.zip                       AN26-358.LMP   FAIL
    pwads/hr/fast/hf142710.zip                        hf142710.lmp   FAIL
    pwads/hr/fast/hf19-532.zip                        HF19-532.LMP   FAIL
    pwads/hr/fast/hf242245.zip                        hf242245.lmp   FAIL
    pwads/hr/fast/hf243019.zip                        HF243019.LMP   FAIL
    pwads/hr/fast/hf32-745.zip                        hf32-745.lmp   FAIL
    pwads/hr/max/hr131314.zip                         hr131314.lmp   FAIL
    pwads/hr/max/hr151233.zip                         hr151233.lmp   FAIL
    pwads/hr/max/hr241315.zip                         hr241315.lmp   FAIL
    pwads/hr/max/hr241545.zip                         hr241545.lmp   FAIL
    pwads/hr/max/hr241626.zip                         hr241626.lmp   FAIL
    pwads/hr/max/hr263309.zip                         hr263309.lmp   FAIL
    pwads/hr/max/hr264105.zip                         hr264105.lmp   FAIL
    pwads/hr/max/hr264502.zip                         hr264502.lmp   FAIL
    pwads/hr/max/hr30-140.zip                         hr30-140.lmp   FAIL
    pwads/hr/movie/2100hr01.zip                       2100hr01.lmp   FAIL
    pwads/hr/movie/3909hr01.zip                       3909hr01.lmp   FAIL
    pwads/hr/respawn/he30-053.zip                     he30-053.lmp   FAIL
    pwads/mm/speed/mm09-512.zip                       mm09-512.lmp   FAIL
    pwads/mm2/fast/m2261438.zip                       M2261438.LMP   FAIL
    pwads/mm2/fast/m232f206.zip                       m232f206.lmp   FAIL
    pwads/mm2/max/m232-143.zip                        m232-143.lmp   FAIL
    pwads/mm2/nmare/m232n037.zip                      m232n037.lmp   FAIL
    pwads/mm2/respawn/m230r257.zip                    m230r257.lmp   FAIL
    pwads/mm2/speed/m230-104.zip                      m230-104.lmp   FAIL
    pwads/mm2/speed/m230-110.zip                      m230-110.lmp   FAIL
    pwads/mm2/tyson/m2321735.zip                      m2321735.lmp   FAIL
    pwads/requiem/fast/rf30-209.zip                   rf30-209.lmp   FAIL
    pwads/requiem/speed/rq23-618.zip                  rq23-618.lmp   FAIL
    tnt/max/ev07-959.zip                              EV07-959.LMP   FAIL
    tnt/nmare/ev28-nm.zip                             EV28-NM.LMP    FAIL
    tnt/nmare/tn32-702.zip                            tn32-702.lmp   FAIL

**Comment by fragglet** [comment]
Wonder if this is similar to #547 and somehow caused by 1bcff874c52ac. Seems like the P_RunThinkers() code should be logically identical to the original version now, though.

**Comment by fabiangreffrath** [comment]
Hm, not sure what the problems are with them. I just tested two random demos out of this list and both played through until the end.

**Comment by fragglet** [comment]
Interesting. What OS are you testing on? I'm using OS X Yosemite (10.10)

**Comment by fragglet** [comment]
I should really add a git bisect script to make tracking down bugs like this easier :)

**Comment by fabiangreffrath** [comment]
```
$ uname -a
Linux kff50 4.0.0-1-amd64 #1 SMP Debian 4.0.2-1 (2015-05-11) x86_64 GNU/Linux
```
That's Debian Jessie with a lot of testing, a bit of unstable and some experimental.

**Comment by fabiangreffrath** [comment]
Just tested f3m4-526.lmp, F26-1824.LMP and 30UV2843.LMP without any problems.

**Comment by fragglet** [comment]
Might be my recent upgrade to Yosemite that has broken the tests, then. Perhaps there's something compiler-dependent being triggered by the version of Clang being used.

**Comment by fabiangreffrath** [comment]
Rebuilt with clang 3.6.0, no difference here.

**Comment by fragglet** [comment]
I'm dumb. I still had -zonezero set in my statcheck configuration file, from looking at #534.

With -zonezero removed, the only one of these demos that still desyncs is:

    pwads/mm/speed/mm09-512.zip                       mm09-512.lmp   FAIL

Which was actually fixed in #368. In this case it looks like the correct fix is actually to change the statcheck expected result, because the demo now plays back correctly.

**Comment by fabiangreffrath** [comment]
This is still the most satisfying possible outcome of this issue. ;)
